src/Features/CSharp/Portable/ConvertForToForEach/CSharpConvertForToForEachCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,152 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using System.Threading;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.ConvertForToForEach;
+using Microsoft.CodeAnalysis.CSharp.CodeGeneration;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle.TypeStyle;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.CodeAnalysis.CSharp.ConvertForToForEach
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = nameof(CSharpConvertForToForEachCodeRefactoringProvider)), Shared]
+    internal class CSharpConvertForToForEachCodeRefactoringProvider :
+        AbstractConvertForToForEachCodeRefactoringProvider<
+            StatementSyntax, 
+            ForStatementSyntax, 
+            ExpressionSyntax,
+            MemberAccessExpressionSyntax,
+            TypeSyntax,
+            VariableDeclaratorSyntax>
+    {
+        protected override string GetTitle()
+            => CSharpFeaturesResources.Convert_for_to_foreach;
+
+        protected override SyntaxList<StatementSyntax> GetBodyStatements(ForStatementSyntax forStatement)
+            => forStatement.Statement is BlockSyntax block
+                ? block.Statements
+                : SyntaxFactory.SingletonList(forStatement.Statement);
+
+        protected override bool TryGetForStatementComponents(
+            ForStatementSyntax forStatement,
+            out SyntaxToken iterationVariable, out ExpressionSyntax initializer,
+            out MemberAccessExpressionSyntax memberAccess, out ExpressionSyntax stepValue,
```

Comment:
it's a little confusing to use 'stepValue' for both the value and the expression it's calculated from, consider adding 'expression' to the name or differentiating it in some other way

Reply:
Sure.

Reply:
#resolved